timesketch/lib/tasks.py

```diff
@@ -138,24 +139,36 @@ def build_index_pipeline(file_path, timeline_name, index_name, file_extension):
         task group.
     """
     index_task_class = _get_index_task_class(file_extension)
-    analyzer_task_group = _get_index_analyzers()
+    index_analyzer_group = _get_index_analyzers()
+    sketch_analyzer_group = None
+
+    if sketch_id:
+        sketch_analyzer_group = build_sketch_analysis_pipeline(sketch_id)
 
     index_task = index_task_class.s(
         file_path, timeline_name, index_name, file_extension)
 
     # If there are no analyzers just run the indexer.
-    if not analyzer_task_group:
+    if not index_analyzer_group and not sketch_analyzer_group:
         return index_task
 
-    return chain(index_task, analyzer_task_group)
+    if sketch_analyzer_group:
+        if not index_analyzer_group:
+            return chain(
+                index_task, run_sketch_init.s(), sketch_analyzer_group)
+        else:
```

Comment:
this else is not needed...

if not index_....:
  return chain(....

return chain(.....

is sufficient

Reply:
Fixed.